baseplate/experiments/providers/simple_experiment.py

```diff
@@ -0,0 +1,150 @@
+
+        return "{}.{}.{}".format(id, name, version)
+
+    def __init__(self, id, name, owner, start_ts, stop_ts, config,
+                 enabled=True, **kwargs):
+
+        self.id = id
+        self.name = name
+        self.owner = owner
+        self.num_buckets = 1000
+
+        self.start_ts = start_ts
+        self.stop_ts = stop_ts
+        self.enabled = enabled
+
+        self.bucket_val = config.get("bucket_val", "user_id")
+        self.version = config.get("experiment_version")
+        self.shuffle_version = config.get("shuffle_version")
+
+        self.variants = config.get("variants", [])
+
+        seed_data = {"id": id, "name": name, "shuffle_version": self.shuffle_version}
+        self.seed = config.get("bucket_seed", self.make_seed(seed_data))
+
+        self._validate_variants(self.variants)
+
+    def _get_seed(self):
+        return self.seed
+
+    def get_unique_id(self, **kwargs):
+        if kwargs.get(self.bucket_val):
+            return ":".join(
+                [self.name, self.bucket_val, str(kwargs[self.bucket_val])]
+            )
+        else:
+            return None
+
+    def should_log_bucketing(self):
+        """ Default to true. Override if logging of eligibility events not required. """
+        return True
+
+    def variant(self, **kwargs):
+        lower_kwargs = {k.lower(): v for k, v in iteritems(kwargs)}
+
+        if not self._is_enabled():
+            return None
+
+        variant = self._check_overrides(**lower_kwargs)
+        if variant is not None and variant in self.variants:
+            return variant
+
+        if self.bucket_val not in lower_kwargs:
+            logger.info(
+                "Must specify %s in call to variant for experiment %s.",
+                self.bucket_val,
+                self.name,
+            )
+            return None
+        if lower_kwargs[self.bucket_val] is None:
+            logger.info(
+                "Cannot choose a variant for bucket value %s = %s "
+                "for experiment %s.",
+                self.bucket_val,
+                lower_kwargs[self.bucket_val],
+                self.name,
+            )
+            return None
+
+        if not self._is_targeted(**lower_kwargs):
+            return None
+
+        bucket = self._calculate_bucket(lower_kwargs[self.bucket_val])
+        return self._choose_variant(bucket)
+
+    def _check_overrides(self, **kwargs):
+        """Check if any of the kwargs override the variant. Functionality
+        to be built in the future. For now, overrides are not supported.
+        """
+        return None
+
+    def _is_targeted(self, **kwargs):
+        """Check if user/etc is targeted for this experiment.
+
+        Advanced targeting functionality to be supported in the future.
+        For now, return True.
+        """
+        return True
+
+    def _is_enabled(self, **kwargs):
+        current_ts = time.time()
+
+        return (self.enabled and current_ts >= self.start_ts
```

Comment:
I don't think the experiments should be checking this, this would be handled by the parser who would return a `ForcedVariantExperiment` if any of this is true.

Reply:
See comment above